pkg/openstack/instances.go

```diff
@@ -613,17 +634,47 @@ func nodeAddresses(srv *servers.Server, interfaces []attachinterfaces.Interface,
 	}
 
 	// process the rest
-	type Address struct {
-		IPType string `mapstructure:"OS-EXT-IPS:type"`
-		Addr   string
-	}
-
 	var addresses map[string][]Address
 	err := mapstructure.Decode(srv.Addresses, &addresses)
 	if err != nil {
 		return nil, err
 	}
 
+	// Get the private networks that are not directly connected
+	if len(addresses) < len(allPrivates) {
+		extraPrivates := make(map[string][]Address)
+		// For each private network
+		for k, v := range allPrivates {
+			ok := false
+			// For each address in the private network
+			for _, a := range v {
+				// Check if the address is directly connected
+				for _, v1 := range addresses {
+					for _, a1 := range v1 {
+						if a.Addr == a1.Addr {
+							ok = true
+							break
+						}
+					}
+				}
+			}
+			// All the addresses in the private network are not directly connected
+			// Save the private network
+			if !ok {
```

Comment:
Then this becomes:

```go
if !found {
```

which is much more readable.

Reply:
comment taken